autoload/vital/__vital__/VS/Vim/Window.vim

```diff
@@ -30,14 +30,97 @@ function! s:do(winid, func) abort
   noautocmd keepalt keepjumps call win_gotoid(l:curr_winid)
 endfunction
 
+"
+" scrollable
+"
+function! s:scrollable(win, ...) abort
+  if len(a:000) == 1
+    call s:set_var(a:win, 'scrollable', a:000[0])
+  else
+    return s:get_var(a:win, 'scrollable')
+  endif
+endfunction
+
+"
+" set_var
+"
+function! s:set_var(win, key, value) abort
+  if winheight(a:win) != -1
+    let l:map = getwinvar(a:win, '___VS_Vim_Window', {})
+    let l:map[a:key] = a:value
+    call setwinvar(a:win, '___VS_Vim_Window', l:map)
+  endif
+endfunction
+
+"
+" get_var
+"
+function! s:get_var(win, key) abort
+  if winheight(a:win) != -1
+    let l:map = getwinvar(a:win, '___VS_Vim_Window', {})
+    return get(l:map, a:key, v:null)
+  endif
+  return v:null
+endfunction
+
+"
+" scroll
+"
+" NOTE: We can't test it because it uses timer.
```

Comment:
curious any reason we use timer?

Reply:
Because it will use with mapping.

```
inoremap <expr><C-f> vital#vital#import('VS.Vim.Window').scroll(+4)
inoremap <expr><C-u> vital#vital#import('VS.Vim.Window').scroll(-4)
```

Reply:
You review is insightful to me. I've change the design.